### PR TITLE
Remixai  chat

### PR DESCRIPTION
**Pull request by STetsing**
Introducing RemixAI Chat

A new Chat interface for interacting with your contracts using AI. 

This PR also fixes the Monaco editor show suggestion by enforcing the suggestion location in the editor

### CODE REVIEW

File: libs/remix-ai-core/src/helpers/streamHandler.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import { JsonStreamParser} from '../types/types'
+
+export const HandleSimpleResponse = async (response, 
+    cb?: (streamText: string) => void) => {
+    let resultText = ''
+    const parser = new JsonStreamParser();
+
+    const chunk = parser.safeJsonParse<{ generatedText: string; isGenerating: boolean }>(response);
+    for (const parsedData of chunk) {
+        if (parsedData.isGenerating) {
```
Comment (LianaHus) [inline libs/remix-ai-core/src/helpers/streamHandler.ts:11]: I think you are not using right setup for linter

File: apps/circuit-compiler/src/app/components/container.tsx [hunk 2]
```diff
@@ -74,15 +74,17 @@ export function Container () {
           explain why the error occurred and how to fix it.
           `
         // @ts-ignore
-        await circuitApp.plugin.call('remixAI', 'error_explaining', message)
+        await circuitApp.plugin.call('remixAI' as any, 'chatPipe', 'error_explaining', message)
+        // await circuitApp.plugin.call('remixAI', 'error_explaining', message)
```
Comment (yann300) [inline apps/circuit-compiler/src/app/components/container.tsx:78]: could you remove all the comments?

File: apps/remix-ide/src/app/plugins/remixAIPlugin.tsx [hunk 3]
```diff
@@ -1,43 +1,67 @@
...
+    } else {
+      console.log('Activating RemixAIPlugin on browser')
+      this.initialize()
+    }
+    this.setRemixAIOnSidePannel(false)
+  }
+
+  setRemixAIOnSidePannel(resize:boolean=false){
+    if (resize){
+      this.call('sidePanel', 'pinView', profile)
```
Comment (yann300) [inline apps/remix-ide/src/app/plugins/remixAIPlugin.tsx:60]: `if` and `else` are doing the same thing.

**Comment by yann300** [comment]
when you started receiving responses from the model, how do you get back to the plugin front page?

File: apps/remix-ide/src/app/plugins/remixAIPlugin.tsx (hunk 3 above)
Comment (STetsing) [reply apps/remix-ide/src/app/plugins/remixAIPlugin.tsx:60]: Waiting for some answers from David to resize the sidepanel

File: apps/remix-ide/src/app/plugins/remixAIPlugin.tsx [hunk 4]
```diff
@@ -161,9 +176,44 @@ export class RemixAIPlugin extends Plugin {
     }
   }
 
-  // render() {
```
Comment (LianaHus) [inline apps/remix-ide/src/app/plugins/remixAIPlugin.tsx:164]: need those?

**Comment by STetsing** [comment]
> when you started receiving responses from the model, how do you get back to the plugin front page?

@yann300 for the moment you can't go back. We can only reinitialize the chat to be empty, which will display the plug-in front page

File: apps/remix-ide/src/app/plugins/remixAIPlugin.tsx (hunk 4 above)
Comment (STetsing) [reply apps/remix-ide/src/app/plugins/remixAIPlugin.tsx:164]: This goes out from this PR

**Review by yann300 — APPROVED** [review]

File: apps/remix-ide/src/app/plugins/remixAIPlugin.tsx (hunk 3 above)
Comment (LianaHus) [reply apps/remix-ide/src/app/plugins/remixAIPlugin.tsx:65]: we will remove it from there anyway, you don't need to do this @STetsing

**Pull request merged**